src/wp-includes/media.php

```diff
@@ -987,9 +984,111 @@ function wp_get_attachment_image_src( $attachment_id, $size = 'thumbnail', $icon
 	 * @param int          $attachment_id Image attachment ID.
 	 * @param string|int[] $size          Requested image size. Can be any registered image size name, or
 	 *                                    an array of width and height values in pixels (in that order).
-	 * @param bool         $icon          Whether the image should be treated as an icon.
 	 */
-	return apply_filters( 'wp_get_attachment_image_src', $image, $attachment_id, $size, $icon );
+	return apply_filters( 'wp_get_attachment_preview_src', $image, $attachment_id, $size );
+}
+
+/**
+ * Retrieves an image to represent an attachment.
+ *
+ * @since 2.5.0
+ *
+ * @param int          		$attachment_id Image attachment ID.
+ * @param string|int[] 		$size          Optional. Image size. Accepts any registered image size name, or an array of
+ *                                    	   width and height values in pixels (in that order). Default 'thumbnail'.
+ * @param bool|string|array $args          {
+ *     Optional. An array of arguments to determine the image retrieved. Default empty array.
```

Comment:
There also should be a note here that for backward compatibility you can also pass a boolean which will be interpreted as value for the `$icon` field.